### ISSUE TITLE
Only highlight perl 6 snippets

### ISSUE THREAD

**Issue by coke** [perl6/doc#1430]
code blocks now can have an explicit :lang<> set; if this lang is not Perl 6, then we should not run the Perl 6 syntax highlighter on those snippets of code. (See, for example, https://docs.perl6.org/language/rb-nutshell, which has many sections that are marked as :lang<ruby>, and therefore should not get Perl 6 highlighting)

**Comment by AlexDaniel** [comment]
I wonder, why can't we highlight ruby blocks as ruby code?

**Comment by coke** [comment]
Regarding highlighting other blocks in their appropriate languages: ... Sure. It's more work, and if someone wants to do it, more power to them.

**Comment by W4anD0eR96** [comment]
relevant code snip

https://github.com/perl6/doc/blob/c1a8cd230e807156268c265093b1e8e77b2c1184/htmlify.p6#L929-L955

BTW, how can we get `:lang<..>` in `htmlify.p6`?

**Comment by coke** [comment]
FWIW, here's the snippet in the examples test that checks whether something is p6 or not:

https://github.com/perl6/doc/blob/master/xt/examples-compilation.t#L50

**Comment by JJ** [comment]
Is this a general recommendation or are there specific snippets where it's happening?

**Comment by AlexDaniel** [comment]
@JJ yes, all “X to Perl 6” pages are affected. See this as an example: https://docs.perl6.org/language/haskell-to-p6#Ranges (Haskell comments not highlighted as comments).
